src/storage/src/hummock/time_travel_version_cache.rs

```diff
@@ -13,43 +13,65 @@
 // limitations under the License.
 
 use std::future::Future;
+use std::pin::Pin;
 
+use ahash::HashMap;
+use futures::future::Shared;
+use futures::FutureExt;
 use moka::sync::Cache;
+use parking_lot::{Mutex, RwLock};
 use risingwave_hummock_sdk::HummockEpoch;
-use tokio::sync::Mutex;
 
 use crate::hummock::local_version::pinned_version::PinnedVersion;
 use crate::hummock::HummockResult;
 
+type InflightRequest = Shared<Pin<Box<dyn Future<Output = HummockResult<PinnedVersion>> + Send>>>;
```

Comment:
Great to know this useful `Shared` future.

With this `Shared` future, I think we don't need to separate the ready `PinnedVersion` and the `InflightRequest`. They can be unified to be having a single `moka::sync::Cache<(TableId, HummockEpoch), InflightResult>>`.

The logic can be as simple as 
```
pub async fn get_or_insert(
    &self,
    table_id: u32,
    epoch: HummockEpoch,
    fetch: impl Future<Output = HummockResult<PinnedVersion>> + Send + 'static,
) -> HummockResult<PinnedVersion> {
    self.cache.entry((epoch, table_id)).or_insert_with(|| fetch.boxed().shared()).value().clone().await
}
```

Reply:
Thanks.